sbi/inference/potentials/score_based_potential.py

```diff
@@ -217,7 +205,7 @@ def rebuild_flow(
         x_density_estimator = reshape_to_batch_event(
             self.x_o, event_shape=self.score_estimator.condition_shape
         )
-        assert x_density_estimator.shape[0] == 1, (
+        assert x_density_estimator.shape[0] == 1 or not self.x_is_iid, (
```

Comment:
Since the `score_iid` PR is almost ready, this was probably changed there. So maybe ping @manuelgloeckler that this might need to be rebased again